Fix #15, error with latest StaticArrays due to dot change.

tkoolen opened the pull request:
CC: @goretkin.

Microbenchmark results: on Julia 1.1.1:

```julia
julia> @btime transpose(a) * b setup = begin
           a = rand(SVector{2})
           b = SVector(rand(SVector{2}), rand(SVector{2}))
           end
  10.521 ns (0 allocations: 0 bytes)
```

which at least doesn't allocate, but on latest master:

```julia
julia> @btime transpose(a) * b setup = begin
           a = rand(SVector{2})
           b = SVector(rand(SVector{2}), rand(SVector{2}))
           end
  1.762 ns (0 allocations: 0 bytes)
```

I think the difference is that an assertion error is being 'outlined' automatically in newer versions. I think this is the cleanest solution going forward, but if you consider this to be too much of a performance bug on 1.1, I can change it.

tkoolen committed:
Fix #15, error with latest StaticArrays due to dot change.

goretkin commented:
Thanks! Did you bump the StaticArrays version because this change isn't compatible with older versions?

I was going to suggest first restricting StaticArrays at v0.5.1-v0.10, and then applying a change that makes it compatible with v0.11, but it seems that Pkg does not support version ranges yet: https://github.com/JuliaLang/Pkg.jl/issues/843

tkoolen commented:
Good point, I guess bumping the StaticArrays lower bound isn't necessary. Pkg does support version ranges, just not spelled that way. The 'correct' thing to do is to retroactively upper bound the StaticArrays version on previous releases, but if we just release a new version with the fix then just doing a `pkg> up` will be fine.

I'm going to replace this implementation with a generated function to get the same performance on 1.1 later today. I had trouble getting down to the performance on master without using a generated function.

Also, not sure what's going on on Julia 1.0 Travis builds.

goretkin commented:
> Pkg does support version ranges, just not spelled that way

Do you mean like: https://github.com/JuliaLang/Pkg.jl/pull/1232#issuecomment-503188542 , or something else?

tkoolen committed:
Switch to generated function approach.

tkoolen commented:
OK, switched to the generated function approach, which is fast for both current and future versions of Julia.

As for the Travis failures on Julia 1, I can reproduce them locally with Julia 1.0.4. *only with code coverage turned on*! So I think this is another instance of https://github.com/JuliaLang/julia/issues/30872.

tkoolen commented:
@rdeits, would you be OK with allowing failures on 1.0?

tkoolen committed:
Allow failures on 1.0 for now.

Because of
https://github.com/JuliaRobotics/EnhancedGJK.jl/pull/16#issuecomment-503319584.

tkoolen commented:
Went ahead with allowing failures on 1.0.

tkoolen commented:
@goretkin, I started the process for tagging a new version: https://github.com/JuliaRegistries/General/pull/1483.
